src/kOS/Utilities/VesselUtils.cs

```diff
@@ -121,13 +121,35 @@ public static double GetMaxThrust(Vessel vessel)
                     if (enginefx != null)
                     {
                         if (!enginefx.isOperational) continue;
-                        thrust += enginefx.maxThrust;
+                        float flowMod = (float)(engine.part.atmDensity / 1.225f);
+                        float velMod = 1.0f;
+                        if (enginefx.atmChangeFlow && enginefx.atmCurve != null)
+                        {
+                            flowMod = enginefx.atmCurve.Evaluate((float)(enginefx.part.atmDensity / 1.225));
+                        }
+                        if (enginefx.velCurve != null)
+                        {
+                            velMod = velMod * enginefx.velCurve.Evaluate((float)vessel.mach);
+                        }
+                        // thrust is fuel flow rate times isp time g times the velocity modifier for jet engines (as of KSP 1.0)
+                        thrust += enginefx.maxFuelFlow * flowMod * enginefx.atmosphereCurve.Evaluate((float)enginefx.part.staticPressureAtm) * enginefx.g * velMod;
                     }
-
-                    if (engine != null)
+                    else if (engine != null)
```

Comment:
When I first found all of the equations, I was reading twice the thrust shown on the right click menu (as near to exactly double as I could see).  It looks like it was the result of counting the thrust twice, once as EngineModulesFx and again as EngineModules.  EngineModulesFx is now a subclass of EngineModules, and it must not have been before.  My first solution was to insert an `else` in the 2nd `if` statement so that it doesn't count it if the 2nd time.  In final practice, we should see if there is any issue just using the EngineModules base class to get this information.  To that same end, we could write it as an extension method of the EngineModules class for MaxThrust and AvailableThrust, which can then be used anywhere else that the thrust needs to be calculated (and also means we only need to edit once if the math ever changes).